### ISSUE TITLE
ConfigurationsErrorException: Error initializing the configuration system

### ISSUE THREAD

**Issue by sebastianhaas** [couchbase/couchbase-lite-net#501]
Hey!

I've encountered this exception while trying to run a unit test suite (NUnit in Xamarin Studio) on a Mac today. The project runs just fine on my Windows machine.

I am using Autofac to resolve the couchbase.lite implementation of the db layer; just ignore the respective lines in the strack trace. The important part I guess is the ``ThreadAbortException`` in ``System.Configuration.ClientConfigurationSystem.get_Configuration ()``. It originates somewhere in the Couchbase Manager configuration setup.

While in debug mode, where I suppose the timeout limit is much higher or indefinite, I found that the execution flow stops in ``Manager.SharedInstance.GetDatabase("databasename");``. Whenever I manually pause the application there, it's doing some weird XML processing.

Any ideas on this?

Thank you very much in advance!

Stack trace: https://gist.github.com/sebastianhaas/992c5eee016bca064341
Configuration:
Mac OSX 10.10.5
Xamarin Studio 5.9.5
Couchbase.Lite 1.1.1.1
SQLite raw & ugly 0.8.2

**Comment by sebastianhaas** [comment]
Update: It seems to be working with 1.1.0.1. Although my tests run absurdly slow.

**Comment by borrrden** [comment]
This looks like an environment issue to me because the manager doesn't do anything even remotely resembling the stack trace that you have shown.  Can you try either:

A) Reproducing this without using Autofac

or

B) Build from source so you can step into the Manager static constructor to see which line gives you the error.

Thanks.

**Comment by sebastianhaas** [comment]
Thanks for the answer.

I bypassed Autofac, leaving me with the following exception:
https://gist.github.com/sebastianhaas/c979a5e9de3669921531

Thanks.

**Comment by sebastianhaas** [comment]
I built it from source now. Takes up to one minute to create an empty database document.
Something is clearly wrong here, but I as don't have the time to look into it I'll close the issue now. Thank you anyway.

**Comment by borrrden** [comment]
Let me know if you find anything else.  I have seen the error you mentioned before but I am not sure what causes it.  I will check the Nuget package and make a new build of it if it is broken.

**Comment by sebastianhaas** [comment]
I just found it strange that obviously the dependencies for .NET 3.5 were resolved although my project was set to 4.5.
I guess I'll head over to the forums for further questions.
On Sep 5, 2015 12:12 AM, Jim Borden <notifications@github.com> wrote:Let me know if you find anything else.  I have seen the error you mentioned before but I am not sure what causes it.  I will check the Nuget package and make a new build of it if it is broken.
